src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerConnectReplyStream.java

```diff
@@ -315,7 +316,13 @@ private void processData(
         {
             final OctetsFW payload = data.payload();
             acceptState.acceptReplyBudget -= payload.sizeof() + acceptState.acceptReplyPadding;
-            assert acceptState.acceptReplyBudget >= 0;
+            String assertionErrorMessage = String.format("[%016x] %s acceptReplyBudget=%d, payload=%d, acceptReplyPadding=%d",
+                                                    currentTimeMillis(),
+                                                    this.toString(),
+                                                    acceptState.acceptReplyBudget,
+                                                    payload.sizeof(),
+                                                    acceptState.acceptReplyPadding);
+            assert acceptState.acceptReplyBudget >= 0 : assertionErrorMessage;
```

Comment:
The assertion message will be continuously constructed as each `DATA` frame arrives, which is probably too much overhead, especially as it creates garbage with formatting the string.

How about we put this inside an `if (acceptState.acceptReplyBudget < 0)` instead?